### PR TITLE
Fix raw deployment service port

### LINKED ISSUE #2961: RawDeployments do not work with GRPC examples
/kind bug

**What steps did you take and what happened:**
It seems that RawDeployment mode with GRPC does not work. Using RawDeployment mode and https://kserve.github.io/website/0.10/modelserving/v1beta1/transformer/torchserve_image_transformer/#run-a-prediction with the following inference service:

```yaml
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: ts-grpc
  namespace: kserve-demo
spec:
  predictor:
    triton:
      storageUri: gs://kfserving-examples/models/torchscript
      runtimeVersion: 20.10-py3
      ports:
        - containerPort: 9000
          name: h2c
          protocol: TCP
      env:
        - name: OMP_NUM_THREADS
          value: "1"
```

Creates the following ingress object

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
...
spec:
  ingressClassName: openshift-default
  rules:
  - host: torchscript-grpc-kserve-demo.apps.rlehmann-ocp-4-12.serverless.devcluster.openshift.com
    http:
      paths:
      - backend:
          service:
            name: torchscript-grpc-predictor
            port:
              number: 80
        path: /
        pathType: Prefix
  - host: torchscript-grpc-predictor-kserve-demo.apps.rlehmann-ocp-4-12.serverless.devcluster.openshift.com
    http:
      paths:
      - backend:
          service:
            name: torchscript-grpc-predictor
            port:
              number: 80
        path: /
        pathType: Prefix
```

Pointing to port 80, while the k8s service looks like this. Note the port 9000 there:

```yaml
apiVersion: v1
kind: Service
metadata:
  name: ts-grpc-predictor
  namespace: kserve-demo
spec:
...
  ports:
  - name: h2c
    port: 9000
    protocol: TCP
    targetPort: 9000
  selector:
    app: isvc.ts-grpc-predictor
  sessionAffinity: None
  type: ClusterIP
```

**What did you expect to happen:**
GRPC port should be propagated to the ingress object accordingly.

**What's the InferenceService yaml:**
See above

**Anything else you would like to add:**
No

**Environment:**

- Istio Version: not related
- Knative Version: not applicable
- KServe Version: 0.10.1
- Kubeflow version: not applicable
- Cloud Environment: none
- Minikube/Kind version: none
- Kubernetes version: not related
- OS: not related

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2961 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by yuzisun** [comment]
@sivanantha321 Any idea how this [grpc test for raw deployment](https://github.com/kserve/kserve/blob/master/test/e2e/predictor/test_raw_deployment.py#L115) works before your change  ?

**Comment by sivanantha321** [comment]
> @sivanantha321 Any idea how this [grpc test for raw deployment](https://github.com/kserve/kserve/blob/master/test/e2e/predictor/test_raw_deployment.py#L115) works before your change ?

@yuzisun As you can see that we are providing container ports in the test. The previous code base uses the provided container port to expose both the [service](https://github.com/kserve/kserve/blob/1314130e93375bbda6ba46703af07a52262d1399/pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go#L75) and [ingress](https://github.com/kserve/kserve/blob/1314130e93375bbda6ba46703af07a52262d1399/pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go#L224). But this only work for the custom containers.

File: pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go [hunk 1]
```diff
@@ -220,11 +220,7 @@ func createRawIngress(scheme *runtime.Scheme, isvc *v1beta1.InferenceService,
 			return nil, fmt.Errorf("failed creating explainer ingress host: %v", err)
 		}
 		// :predict routes to the predictor when there is only predictor and explainer
-		if len(isvc.Spec.Predictor.Containers) != 0 && len(isvc.Spec.Predictor.Containers[0].Ports) != 0 {
-			rules = append(rules, generateRule(host, predictorName, "/", isvc.Spec.Predictor.Containers[0].Ports[0].ContainerPort))
-		} else {
-			rules = append(rules, generateRule(host, predictorName, "/", constants.CommonDefaultHttpPort))
-		}
+		rules = append(rules, generateRule(host, predictorName, "/", constants.CommonDefaultHttpPort))
```
Comment (yuzisun) [inline pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go:223]: I see so `if len(isvc.Spec.Predictor.Containers) != 0 && len(isvc.Spec.Predictor.Containers[0].Ports) != 0` only works for custom predictor, for standard predictor you would need to check the ports on the `model` spec.

**Comment by yuzisun** [comment]
> > @sivanantha321 Any idea how this [grpc test for raw deployment](https://github.com/kserve/kserve/blob/master/test/e2e/predictor/test_raw_deployment.py#L115) works before your change ?
> 
> @yuzisun As you can see that we are providing container ports in the test. The previous code base uses the provided container port to expose both the [service](https://github.com/kserve/kserve/blob/1314130e93375bbda6ba46703af07a52262d1399/pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go#L75) and [ingress](https://github.com/kserve/kserve/blob/1314130e93375bbda6ba46703af07a52262d1399/pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go#L224). But this only work for the custom containers.

Got it, another side issues I found about the ingress host name which still contains the `default` suffix, not sure which host we are using in the test.

```
  spec:
    ingressClassName: istio
    rules:
    - host: sklearn-iris-grpc-default.example.com
      http:
        paths:
        - backend:
            service:
              name: sklearn-iris-grpc-predictor
              port:
                number: 80
          path: /
          pathType: Prefix
    - host: sklearn-iris-grpc-predictor-default.example.com
      http:
        paths:
        - backend:
            service:
              name: sklearn-iris-grpc-predictor
              port:
                number: 80
          path: /
          pathType: Prefix
```

File: pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go (hunk 1 above)
Comment (sivanantha321) [reply pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go:223]: indeed

**Comment by sivanantha321** [comment]
> > > @sivanantha321 Any idea how this [grpc test for raw deployment](https://github.com/kserve/kserve/blob/master/test/e2e/predictor/test_raw_deployment.py#L115) works before your change ?
> > 
> > 
> > @yuzisun As you can see that we are providing container ports in the test. The previous code base uses the provided container port to expose both the [service](https://github.com/kserve/kserve/blob/1314130e93375bbda6ba46703af07a52262d1399/pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go#L75) and [ingress](https://github.com/kserve/kserve/blob/1314130e93375bbda6ba46703af07a52262d1399/pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go#L224). But this only work for the custom containers.
> 
> Got it, another side issues I found about the ingress host name which still contains the `default` suffix, not sure which host we are using in the test.
> 
> ```
>   spec:
>     ingressClassName: istio
>     rules:
>     - host: sklearn-iris-grpc-default.example.com
>       http:
>         paths:
>         - backend:
>             service:
>               name: sklearn-iris-grpc-predictor
>               port:
>                 number: 80
>           path: /
>           pathType: Prefix
>     - host: sklearn-iris-grpc-predictor-default.example.com
>       http:
>         paths:
>         - backend:
>             service:
>               name: sklearn-iris-grpc-predictor
>               port:
>                 number: 80
>           path: /
>           pathType: Prefix
> ```

I believe `default` refers to namespace here not the suffix.

**Pull request merged**